velum.yaml

```diff
@@ -52,26 +51,26 @@ spec:
       value: worker1
     - name: RAILS_ENV
       value: production
-    - name: SECRET_KEY_BASE
-      value: fc0ece3811c0b1e0118dd7d619b385955e3ed146b37315d67d1c2d201c42531da398941ef27aab42493fff23afcba540390a26db51b5889e96d850a14178aabdc1926c013936a3c41abdbeb696ce440685c5d7be3a6c548434699d526a9e37657a76a43ca975b824905fd949f7e8d50aa1196a58e7e79dcfb08e2e397ee4f7b7 # FIXME: change for production (we should autogenerate this for every installation and use the autogenerated value)
     - name: VELUM_DB_HOST
       value: "127.0.0.1"
     - name: VELUM_DB_PORT
       value: "3306"
-    - name: VELUM_DB_USERNAME
-      value: "root" # FIXME: do not use root user
-    - name: VELUM_DB_PASSWORD
-      value: "salt"
-    - name: VELUM_SALT_HOST
-      value:
-    - name: VELUM_SALT_PORT
-      value:
-    - name: VELUM_SALT_USER
-      value:
+    volumeMounts:
+      - mountPath: /usr/src/app/config/secrets.yml
+        name: secrets-config
+      - mountPath: /usr/src/app/config/database.yml
+        name: database-config
     command: ["/bin/sh","-c"]
     args: ["bundle exec bin/rake salt:process"]
-  - name: velum-mariadb # FIXME: being on the same pod (with hostNetwork: true) also exposes mysql (if it listens on 0.0.0.0).
+  - name: velum-mariadb
     image: sles12/mariadb:10.0
     env:
     - name: MYSQL_ROOT_PASSWORD # FIXME: do not set root password, create user/password
       value: "salt"             # FIXME
+  volumes:
+    - name: secrets-config
+      hostPath:
+        path: SOMEWHERE/secrets.yml # FIXME: define SOMEWHERE
+    - name: database-config
+      hostPath:
+        path: SOMEWHERE/database.yml # FIXME: define SOMEWHERE
```

Comment:
the docker image we use for MariaDB accepts an environment variable (MYSQL_ROOT_PASSWORD) which sets (or resets) the root password to the specified value. We can also use an environment variable in the database.yml file of velum to set the password used. The only question I think is which script will generate the value for those env variables so it is unique among the different installations.